### PR TITLE
Test of gh-actions

### PR DESCRIPTION
**Pull request by MarcusZuber**
Todo:

- [x] Add codecov
- [x] Test on multiple versions of python

### CODE REVIEW

**Comment by tfarago** [comment]
Pretty much the same deal as with circleci, do you have compelling arguments? One for circleci is the 6000 min.

**Comment by MarcusZuber** [comment]
As you prefer. I just wanted to try it.

**Comment by tfarago** [comment]
Actions are actually free for public repos, so gonna switch to this because of unlimited build time.

**Comment by MarcusZuber** [comment]
Note: python3.10-test requires https://github.com/ufo-kit/concert/pull/490

**Comment by tfarago** [comment]
Why do you have a separate coverage.yml now?

**Comment by MarcusZuber** [comment]
Then it was easier to run coverage only for one python version.

**Comment by tfarago** [comment]
So it would call it as many times as you have python versions?

**Comment by MarcusZuber** [comment]
The current test.yml iterates over this list of python versions. For sure it would be possible to tell it to do the coverage only once, but splitting it was more simple.

**Comment by tfarago** [comment]
I think the if statement is more simple than a whole new file, plus you use more minutes because you test one more time then necessary, all you need it `if: matrix.python-version == 3.8` in the coverage setup, check [this](https://github.com/ufo-kit/syris/blob/ci/.github/workflows/tests.yml#L36-L44) out.

**Comment by MarcusZuber** [comment]
OK, codecov was more clever than I expected.

From my side we can merge.

The python3.10 I removed until we migrate from nose. My fix depends on the import order, which is different for coverage, nosetest and nose.main() :roll_eyes:

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.9]
```
Comment (tfarago) [inline .github/workflows/tests.yml:18]: Why drop 3.10?

**Comment by tfarago** [comment]
You forgot to change the badge in README

File: .github/workflows/tests.yml (hunk 1 above)
Comment (MarcusZuber) [reply .github/workflows/tests.yml:18]: The python3.10 I removed until we migrate from nose. My fix depends on the import order, which is different for coverage, nosetest and nose.main() 🙄

**Comment by MarcusZuber** [comment]
I totally forgot the badge. I will fix this tomorrow.

**Pull request merged**